File: 0071-simplify-path/0071-simplify-path.java
class Solution {
    public String simplifyPath(String path) {
        String[] patharr = path.split("/");
        Stack<String> st = new Stack<>();
        for (String i : patharr) {
            if (i.equals("") || i.equals(".")) {
                continue;
            }
            if (i.equals("..")) {
                if (!st.isEmpty()) {
                    st.pop();
                }
            } else {
                st.push(i);
            }
        }

        StringBuilder sb = new StringBuilder();
        while (!st.isEmpty()) {
            sb.insert(0,"/" + st.pop());
        }

        return sb.length() == 0 ? "/" : sb.toString();        
    }
}

File: 0071-simplify-path/0071-simplify-path.cs
public class Solution {
    public string SimplifyPath(string path) {
        string[] str=path.Split('/');
        StringBuilder sb=new StringBuilder();
        Stack<string> st=new Stack<string>();
        for(int i=0;i<str.Length;i++){
            if(str[i]=="" || str[i]=="." ){
                continue;
            }
            else if(str[i]==".." ){
               if(st.Count!=0) st.Pop();
            }
            else{
                st.Push(str[i]);
            }
        }
        if(st.Count==0)return "/";
        while(st.Count>0){
            if(st.Count!=0){
                sb.Insert(0,st.Pop());
                sb.Insert(0,"/");
            }
        }
        return sb.ToString();
    }
}